test/integration/sonobuoy_integration_test.go

```diff
@@ -46,15 +49,43 @@ func runSonobuoyCommandWithContext(ctx context.Context, t *testing.T, args strin
 
 	t.Logf("Running %q\n", command.String())
 
-	return command.Run(), stdout, stderr
+	return stdout, stderr, command.Run()
+}
+
+func mustRunSonobuoyCommand(t *testing.T, args string) bytes.Buffer {
+	return mustRunSonobuoyCommandWithContext(context.Background(), t, args)
+}
+
+// mustRunSonobuoyCommandWithContext runs the Sonobuoy CLI with the given context and arguments.
+// It returns stdout and fails the test immediately if there are any errors.
+func mustRunSonobuoyCommandWithContext(ctx context.Context, t *testing.T, args string) bytes.Buffer {
+	var stdout, stderr bytes.Buffer
+
+	command := exec.CommandContext(ctx, sonobuoy, strings.Fields(args)...)
+	command.Stdout = &stdout
+	command.Stderr = &stderr
+
+	t.Logf("Running %q\n", command.String())
+	if err := command.Run(); err != nil {
+		t.Fatalf("Expected %q to not error but got error: %q with stdout: %q and stderr: %q", args, err, stdout.String(), stderr.String())
+	}
+
+	return stdout
 }
 
 // runSonobuoyCommand runs the Sonobuoy CLI with the given arguments and a background context.
 // It returns any encountered error and the stdout and stderr from the command execution.
-func runSonobuoyCommand(t *testing.T, args string) (error, bytes.Buffer, bytes.Buffer) {
+func runSonobuoyCommand(t *testing.T, args string) (bytes.Buffer, bytes.Buffer, error) {
 	return runSonobuoyCommandWithContext(context.Background(), t, args)
 }
 
+// getNamespace returns the namespace to use for the current test and a function to clean it up
+// asynchronously afterwards.
+func getNamespace(t *testing.T) (string, func()) {
+	ns := "sonobuoy-" + strings.ToLower(t.Name())
+	return ns, func() { cleanup(t, ns) }
+}
```

Comment:
Expanded the helper; cleanup was being used throughout but the boilerplate for each test was a bit verbose. This simplifies the calls.